packages/cactus-test-geth-ledger/src/test/typescript/integration/geth-test-ledger.test.ts

```diff
@@ -0,0 +1,184 @@
+/**
+ * Tests of Geth helper typescript setup class.
+ *
+ * TODO:
+ *    - Refactor to use `Clef` instead of `eth.personal`
+ */
+
+//////////////////////////////////
+// Constants
+//////////////////////////////////
+
+// Ledger settings
+// const containerImageName = "ghcr.io/hyperledger/cactus-geth-all-in-one";
+// const containerImageVersion = "2022-10-18-06770b6c";
+// const useRunningLedger = false;
+
+// Log settings
+const testLogLevel: LogLevelDesc = "info";
+
+import { GethTestLedger } from "../../../main/typescript/index";
+import contractData from "../../typescript/solidity/hello-world-contract/HelloWorld.json";
+
+import {
+  LogLevelDesc,
+  LoggerProvider,
+  Logger,
+} from "@hyperledger/cactus-common";
+import { pruneDockerAllIfGithubAction } from "@hyperledger/cactus-test-tooling";
+
+import "jest-extended";
+import { Web3 } from "web3";
+
+// Logger setup
+const log: Logger = LoggerProvider.getOrCreate({
+  label: "geth-test-ledger.test",
+  level: testLogLevel,
+});
+
+/**
+ * Main test suite
+ */
+
+describe("Geth Test Ledger checks", () => {
+  let ledger: GethTestLedger;
+  let testEthAcc: any;
+  let web3Instance: Web3;
+  //////////////////////////////////
+  // Environment Setup
+  //////////////////////////////////
+
+  beforeAll(async () => {
+    log.info("Prune Docker...");
+    await pruneDockerAllIfGithubAction({ logLevel: testLogLevel });
+
+    log.info("Start GethTestLedger...");
+    ledger = new GethTestLedger({
+      emitContainerLogs: true,
+      logLevel: testLogLevel,
+    });
+    expect(ledger).toBeTruthy();
+    log.debug("Geth image:", ledger.fullContainerImageName);
+
+    await ledger.start(true);
```

Comment:
@rwat17 Omitting the image pull by default will make the test case permanently broken